wpimath/src/main/native/include/frc/system/NumericalIntegration.h

```diff
@@ -237,4 +241,178 @@ T RKDP(F&& f, units::second_t t, T y, units::second_t dt,
   return y;
 }
 
+/**
+ * Performs adaptive Tsitouras 5th Order integration of dx/dt = f(x, u) for dt.
+ *
+ * @param f        The function to integrate. It must take two arguments x and
+ *                 u.
+ * @param x        The initial value of x.
+ * @param u        The value u held constant over the integration period.
+ * @param dt       The time over which to integrate.
+ * @param maxError The maximum acceptable truncation error. Usually a small
+ *                 number like 1e-6.
+ */
+template <typename F, typename T, typename U>
+T Tsit5(F&& f, T x, U u, units::second_t dt, double maxError = 1e-6) {
+  // See http://users.uoa.gr/~tsitourasc/RK54_new_v2.pdf for the
+  // Butcher tableau the following arrays came from.
+
+  constexpr int kDim = 7;
+
+  // clang-format off
+  constexpr double A[kDim - 1][kDim - 1]{
+      {              0.161},
+      {  -0.008480655492357,    0.3354806554923570},
+      {   2.897153057105494,     -6.359448489975075,   4.362295432869581},
+      { 5.32586482843925895,     -11.74888356406283,    7.495539342889836, -0.09249506636175525},
+      { 5.86145544294642038,     -12.92096931784711,    8.159367898576159,  -0.07158497328140100, -0.02826905039406838},
+      { 0.09646076681806523,                   0.01,   0.4798896504144996,     1.379008574103742,    -3.290069515436081, 2.324710524099774}};
```

Comment:
The row here belongs to b, not A. If this implementation is correct, more comments are needed to explain it.

Reply:
a7i = bi, i = 1, 2, ··· , 6